kyo-core/shared/src/main/scala/kyo/seqs.scala

```diff
@@ -4,70 +4,118 @@ import kyo.core.*
 
 object Seqs:
 
-    def traverse[T, U, S, S2](v: Seq[T])(f: T => U < S2): Seq[U] < (S & S2) =
-        collect(v.map(f))
-
-    def traverseUnit[T, U, S](v: Seq[T])(f: T => Unit < S): Unit < S =
-        def loop(l: Seq[T]): Unit < S =
-            if l.isEmpty then ()
-            else f(l.head).andThen(loop(l.tail))
-        loop(v)
-    end traverseUnit
+    def map[T, U, S, S2](seq: Seq[T])(f: T => U < S2): Seq[U] < (S & S2) =
+        seq.size match
+            case 0 => Seq.empty
+            case 1 => f(seq(0)).map(Seq(_))
+            case size =>
+                seq match
+                    case seq: IndexedSeq[T] =>
+                        Loops.indexed(Chunks.init[U]) { (idx, acc) =>
+                            if idx == size then Loops.done(acc.toSeq)
+                            else f(seq(idx)).map(u => Loops.continue(acc.append(u)))
+                        }
+                    case seq: List[T] =>
+                        Loops.transform(seq, Chunks.init[U]) { (seq, acc) =>
+                            seq match
+                                case Nil          => Loops.done(acc.toSeq)
+                                case head :: tail => f(head).map(u => Loops.continue(tail, acc.append(u)))
+                        }
+                    case seq =>
+                        Chunks.initSeq(seq).map(f).map(_.toSeq)
```

Comment:
All methods follow a similar pattern specializing for small collections, indexed seqs, and lists. The "slow path" uses `Chunks`, which we can probably optimize further.